packages/turf-center-median/index.js

```diff
@@ -0,0 +1,124 @@
+ * produces a value that respects that. Unlike `@turf/center`, it is
+ * sensitive to clusters and outliers. It lands in the statistical middle of a
+ * dataset, not the geographical. It can also be weighted, meaning certain
+ * points are more important than others.
+ *
+ * `@turf/center-median` takes the mean center and tries to find, iteratively,
+ * a new point that requires the least amount of travel from all the points in
+ * the dataset. It is not as sensitive to outliers as `@turf/center`, but it is
+ * attracted to clustered data. It, too, can be weighted.
+ *
+ * @name centerMedian
+ * @param {FeatureCollection<Point>} features Feature Collection of GeoJSON Points
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.weight] the property name used to weight the center
+ * @param {number} [options.tolerance=0.001] the difference in distance between candidate medians at which point the algorighim stops iterating.
+ * @param {number} [options.counter=10] how many attempts to find the median, should the tolerance be insufficient.
+ * @returns {Feature<Point>} The median center of the collection
+ * @example
+ * var points = turf.points([[0, 0], [1, 0], [0, 1], [5, 8]]);
+ * var medianCenter = turf.centerMedian(points);
+ *
+ * //addToMap
+ * var addToMap = [points, medianCenter]
+ */
+function centerMedian(features, options) {
+    // Optional params
+    options = options || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var counter = options.counter || 10;
+    if (!isNumber(counter)) throw new Error('counter must be a number');
+    var weightTerm = options.weight;
+
+    // Calculate mean center:
+    var meanCenter = centerMean(features, {weight: options.weight});
+
+    // Calculate center of every feature:
+    var centroids = featureCollection([]);
+    featureEach(features, function (feature) {
+        centroids.features.push(centroid(feature, {weight: feature.properties[weightTerm]}));
+    });
+
+    centroids.properties = {
+        tolerance: options.tolerance,
+        medianCandidates: []
+    };
+    return findMedian(meanCenter, [0, 0], centroids, counter);
+}
+
+/**
+ * Recursive function to find new candidate medians.
+ *
+ * @private
+ * @param {Feature<Point>} candidateMedian current candidate median
+ * @param {Feature<Point>} previousCandidate the previous candidate median
+ * @param {FeatureCollection<Point>} centroids the collection of centroids whose median we are determining
+ * @param {number} counter how many attempts to try before quitting.
+ * @returns {Feature<Point>} the median center of the dataset.
+ */
+function findMedian(candidateMedian, previousCandidate, centroids, counter) {
+    var tolerance = centroids.properties.tolerance || 0.001;
+    var candidateXsum = 0;
+    var candidateYsum = 0;
+    var kSum = 0;
+    var centroidCount = 0;
+    featureEach(centroids, function (theCentroid) {
```

Comment:
Instead of calling it `theCentroid` you can simply use a singular variable name like `centroid` if the input is plural (`centroids`)